### PR TITLE
Migrate components to TS: HelpText

### LINKED ISSUE #19121: Migrate components to TS: HelpText
### Description
Migrate `HelpText` component to TypeScript

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by taesuklee**
## Explanation
Fixes #19121 

Migrate [HelpText components](https://github.com/MetaMask/metamask-extension/tree/develop/ui/components/component-library/help-text) to TypeScript.

## Screenshots/Screencaps

No visual regression after migration.

### Before
Default
image
Children
image
Severity
image
Color
image

### After
Default
image

Children
image

Severity
image

Color
image

## Manual Testing Steps

7 unit tests by `yarn jest ui/components/component-library/help-text/help-text.test.tsx` pass.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by taesuklee** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by taeslee** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/component-library/help-text/help-text.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import classnames from 'classnames';
+import {
+  TextVariant,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import { Text, ValidTag } from '../text';
+import { HelpTextProps, HelpTextSeverity } from './help-text.types';
+
+export const HelpText = ({
```
Comment (georgewrmarshall) [inline ui/components/component-library/help-text/help-text.tsx:10]: I know this wasn't in the original issue but could we please forwardRefs for this component. I think it makes sense to for all of our components

File: ui/components/component-library/help-text/help-text.types.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+export enum HelpTextSeverity {
+  Danger = Severity.Danger,
+  Warning = Severity.Warning,
+  Success = Severity.Success,
+  Info = Severity.Info,
+}
+
+export interface HelpTextProps extends TextProps {
+  severity?: HelpTextSeverity;
+
```
Comment (georgewrmarshall) [inline ui/components/component-library/help-text/help-text.types.ts:14]: nit: remove line break
```suggestion
```

File: ui/helpers/constants/design-system.ts [hunk 3]
```diff
@@ -389,6 +389,18 @@ export const FONT_STYLE = {
...
+}
+
+/**
+ * @deprecated `SEVERITIES` const has been deprecated in favour of the `Severities` enum which can still be imported from `ui/helpers/constants/design-system.ts`
+ *
+ * `SEVERITIES` can be removed once the migration from banner-alert.js  to TS.
+ */
 export const SEVERITIES = {
   DANGER: 'danger',
   WARNING: 'warning',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:406]: You may have to rebase this. The correct enum is `Severity`

**Review by georgewrmarshall — COMMENTED** [review]
Hey @taesuklee, thanks for your contribution this is looking really great!! I've left a few small suggestions and a request which was out of scope of the original ticket but hoping you can accommodate it.

**Pull request closed without merging**